### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by dunyasalah**
Added login and logout functionality

### CODE REVIEW

File: src/models/userModel.js [hunk 1]
```diff
@@ -51,4 +51,9 @@ userSchema.pre('save', async function (next) {
   next();
 });
 
+userSchema.methods.correctPassword = async function(candidatePassword) {
+  return await bcrypt.compare(candidatePassword, this.password);
+};
+
```
Comment (AhmedSalman1) [inline src/models/userModel.js:55]: remove this line

File: src/models/userModel.js [hunk 2]
```diff
@@ -51,4 +51,9 @@ userSchema.pre('save', async function (next) {
   next();
 });
 
+userSchema.methods.correctPassword = async function(candidatePassword) {
```
Comment (AhmedSalman1) [inline src/models/userModel.js]: add space after the keyword "function"

File: src/controllers/authController.js [hunk 3]
```diff
@@ -41,3 +41,30 @@ export const signup = catchAsyncError(async (req, res) => {
...
+
+  // logout
+  export const logout = (req, res) => {
+    res.cookie('jwt', 'loggedout', {
+      expires: new Date(Date.now() + 10 * 1000),  
+      httpOnly: true,
+    });
+  
+    res.status(200).json({ status: 'success' });
+  };
```
Comment (AhmedSalman1) [inline src/controllers/authController.js]: fix this bug

File: src/models/userModel.js (hunk 1 above)
Comment (AhmedSalman1) [inline src/models/userModel.js:55]: remove this line

File: src/controllers/authController.js [hunk 4]
```diff
@@ -41,3 +41,30 @@ export const signup = catchAsyncError(async (req, res) => {
 
   createSendToken(newUser, 201, req, res);
 });
+
+// login
```
Comment (aymanshalaby55) [inline src/controllers/authController.js:52]: make route for this functions

File: src/controllers/authController.js [hunk 5]
```diff
@@ -41,3 +41,30 @@ export const signup = catchAsyncError(async (req, res) => {
...
+  if (!email || !password) {
+    return res.status(400).json({ message: 'Please provide your email and password' });
+  }
+  const user = await User.findOne({ email }).select( '+password' );
+});
+  if (!user || !( await user.correctPassword(password, user.password))) {
+    return res.status(401).send({ message: 'Incorrect email or password' });
+  }
+  reateSendToken(user, 200, req, res);
+
```
Comment (aymanshalaby55) [inline src/controllers/authController.js]: insert these lines in login function

**Review by aymanshalaby55 — COMMENTED** [review]
review

File: src/models/userModel.js [hunk 6]
```diff
@@ -42,18 +42,13 @@ const userSchema = new mongoose.Schema({
...
   this.password = await bcrypt.hash(this.password, 12);
   next();
 });
-
-userSchema.methods.correctPassword = async function(candidatePassword) {
-  return await bcrypt.compare(candidatePassword, this.password);
-};
-
+userSchema.methods.correctPassword = async (candidatePassword) =>
+  await bcrypt.compare(candidatePassword, this.password);
```
Comment (AhmedSalman1) [inline src/models/userModel.js]: don't use arrow function(=>) here (this) doesn't refer to current document    use regular function instead like this :

userSchema.methods.correctPassword = async function (candidatePassword) {
	return await bcrypt.compare(candidatePassword, this.password);
};

File: .husky/_/.gitignore [hunk 7]
```diff
@@ -1,4 +1 @@
-*
-!commit-msg
-!pre-commit
-!husky.sh
```
Comment (AhmedSalman1) [inline .husky/_/.gitignore]: why did you remove these lines?

**Review by aymanshalaby55 — APPROVED** [review]
